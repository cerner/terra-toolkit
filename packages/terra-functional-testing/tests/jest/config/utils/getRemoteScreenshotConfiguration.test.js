jest.mock('fs-extra');
jest.mock('js-yaml');
const mockWarn = jest.fn();
jest.mock('@cerner/terra-cli/lib/utils/Logger', () => function mock() {
  return {
    warn: mockWarn,
  };
});

const fs = require('fs-extra');
const path = require('path');
const yaml = require('js-yaml');
const getRemoteScreenshotConfiguration = require('../../../../src/config/utils/getRemoteScreenshotConfiguration');

const testScreenshotsSites = {
  repositoryId: 'test-repository-id',
  repositoryUrl: 'https://github.com/cerner/terra-toolkit.git',
};

describe('getRemoteScreenshotConfiguration', () => {
  const oldCwd = process.cwd;
  beforeEach(() => {
    mockWarn.mockClear();
  });

  beforeAll(() => {
    process.cwd = jest.fn();
    process.cwd.mockReturnValue('/app');

    fs.readFileSync.mockReturnValue('yaml string');
    yaml.safeLoad.mockReturnValue({
      'test-repository-id': {
        username: 'username',
        password: 'password',
      },
    });
  });

  afterAll(() => {
    process.cwd = oldCwd;
  });

  it('loads the default config', () => {
    fs.readJsonSync.mockReturnValueOnce({
      name: 'terra-functional-testing',
      version: '1.0.0',
      repository: {
        url: 'https://github.com/cerner/terra-toolkit.git',
      },
    });

    const config = getRemoteScreenshotConfiguration(testScreenshotsSites);
    expect(config).toMatchSnapshot();
    expect(fs.readJsonSync).toHaveBeenCalledWith(path.join(process.cwd(), 'package.json'));
  });

  it('loads the config for branch', () => {
    fs.readJsonSync.mockReturnValueOnce({
      name: 'terra-functional-testing',
      version: '1.0.0',
      repository: {
        url: 'https://github.com/cerner/terra-toolkit.git',
      },
    });
    const config = getRemoteScreenshotConfiguration(testScreenshotsSites, 'dev');
    expect(config).toMatchSnapshot();
    expect(fs.readJsonSync).toHaveBeenCalledWith(path.join(process.cwd(), 'package.json'));
  });

  it('throws an error with a bad site', () => {
    const testSites = {
      repositoryId: 'test-repository-id',
      repositoryUrl: 'http://www.badsite.com',
    };

    fs.readJsonSync.mockReturnValueOnce({
      name: 'terra-functional-testing',
      version: '1.0.0',
      repository: {
        url: 'https://github.com/cerner/terra-toolkit.git',
      },
    });
    expect(() => getRemoteScreenshotConfiguration(testSites)).toThrowErrorMatchingSnapshot();
  });
});
