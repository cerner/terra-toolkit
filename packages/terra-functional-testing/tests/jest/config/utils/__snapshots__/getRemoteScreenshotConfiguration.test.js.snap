// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRemoteScreenshotConfiguration loads the config for branch 1`] = `
Object {
  "publishScreenshotConfiguration": Object {
    "referenceScreenshotsPath": "/app/tests/wdio/__snapshots__/reference",
    "serviceAuthHeader": "Basic dXNlcm5hbWU6cGFzc3dvcmQ=",
    "serviceUrl": "https://github.com/cerner/terra-toolkit.git/content-compressed/com.cerner.cerner/terra-functional-testing/build-screenshots/dev/",
    "url": "https://github.com/nexus/content/sites/terra-toolkit.git/com.cerner.cerner/terra-functional-testing/build-screenshots/dev/",
    "zipFilePath": "/app/tests/wdio",
  },
}
`;

exports[`getRemoteScreenshotConfiguration loads the default config 1`] = `
Object {
  "publishScreenshotConfiguration": Object {
    "referenceScreenshotsPath": "/app/tests/wdio/__snapshots__/reference",
    "serviceAuthHeader": "Basic dXNlcm5hbWU6cGFzc3dvcmQ=",
    "serviceUrl": "https://github.com/cerner/terra-toolkit.git/content-compressed/com.cerner.cerner/terra-functional-testing/build-screenshots/master/",
    "url": "https://github.com/nexus/content/sites/terra-toolkit.git/com.cerner.cerner/terra-functional-testing/build-screenshots/master/",
    "zipFilePath": "/app/tests/wdio",
  },
}
`;

exports[`getRemoteScreenshotConfiguration throws an error with a bad site 1`] = `"The repository url is malformed: http://www.badsite.com"`;
